### PR TITLE
Taking the parameter literally

### PR DESCRIPTION
**Pull request by sima-zhu**
# Description

We now have doc validation step in unified pipeline and release pipeline. 
The PR is to implement the validation function in each language.

We also have docindex pipeline for all package validation. Make some changes to make it work.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -336,7 +336,10 @@ function PackageDependenciesResolve($artifactNamePrefix, $packageDirectory) {
   return $true
 }
 
-function ValidatePackage($groupId, $artifactId, $version, $workingDirectory) {
+function ValidatePackage($groupId, $artifactId, $version) {
+  $workingDirectory = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:340]: Do we want to have a unique working directory for each call to this method?

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -610,4 +610,19 @@ function Get-java-DocsMsMetadataForPackage($PackageInfo) {
...
+    [Parameter(Mandatory = $true)]
+    [PSCustomObject]$PackageInfo,
+    [Parameter(Mandatory = $false)]
+    [string]$PackageSourceOverride,
+    [Parameter(Mandatory = $false)]
+    [string]$DocValidationImageId
+  ) 
+  $packageGroupId = $PackageInfo.Name
+  $packageVersion= $PackageInfo.Version
+  $p
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:626]: This looks like it is leftover.

Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:626]: Or you intended to defined packageName.

Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:626]: Why actually create local variables as opposed to just passing them in directly to the command below?

File: eng/scripts/Language-Settings.ps1 [hunk 3]
```diff
@@ -610,4 +610,19 @@ function Get-java-DocsMsMetadataForPackage($PackageInfo) {
...
+  ValidatePackage $packageGroupId $packageName $packageVersion
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:627]:
```suggestion
  ValidatePackage $PackageInfo.GroupId $PackageInfo.Name  $PackageInfo.Version
```

**Review by weshaggard — APPROVED** [review]

File: eng/scripts/Language-Settings.ps1 [hunk 4]
```diff
@@ -610,4 +612,9 @@ function Get-java-DocsMsMetadataForPackage($PackageInfo) {
     PreviewReadMeLocation = 'docs-ref-services/preview'
     Suffix = ''
   }
-}
\ No newline at end of file
+}
+
+function Validate-java-DocMsPackages ($PackageInfo, $DocValidationImageId)
```
Comment (benbp) [inline eng/scripts/Language-Settings.ps1:617]: `$DocValidationImageId` appears unused in both function signatures.

**Comment by benbp** [comment]
Not related to your changes here but a more general comment: as someone reviewing who isn't aware of the context of what you're working on, it would be helpful to have a more descriptive PR title and/or description of the changes (initial issue, intent of the change, problems fixed, decisions made, etc.).

File: eng/scripts/Language-Settings.ps1 (hunk 4 above)
Comment (sima-zhu) [reply eng/scripts/Language-Settings.ps1:617]: It is a placeholder. Our java docker image is broken. I will fix the image first before we adding the real logic

**Comment by sima-zhu** [comment]
> Not related to your changes here but a more general comment: as someone reviewing who isn't aware of the context of what you're working on, it would be helpful to have a more descriptive PR title and/or description of the changes (initial issue, intent of the change, problems fixed, decisions made, etc.).

@benbp Add description as you suggested.

**Pull request merged**